Volumes API

strokine opened the pull request:
Adding support for Volumes API.

jeevatkm commented:
@strokine nice (#54), thanks for PR. I will have a look and get back to you.

strokine commented:
Committed code for Volume Actions as well.

strokine commented:
Completed all Block Storage API.

jeevatkm commented on src/main/java/com/myjeeva/digitalocean/impl/DigitalOceanClient.java:
```diff
@@ -1056,8 +1081,130 @@ public Response untagResources(String name, List<Resource> resources)
+	@Override
+	public Volume createVolume(Volume volume) throws DigitalOceanException, RequestUnsuccessfulException {
+	    if (null == volume
+	        || StringUtils.isEmpty(volume.getName())
```
Please change it to `StringUtils.isBlank()`.

jeevatkm commented on src/main/java/com/myjeeva/digitalocean/pojo/Volume.java:
```diff
@@ -0,0 +1,85 @@
+	private String description;
+	
+	@SerializedName("size_gigabytes")
+	private Integer sizeGigabytes;
```
Please change datatype and variable name like [here](https://github.com/jeevatkm/digitalocean-api-java/blob/master/src/main/java/com/myjeeva/digitalocean/pojo/Image.java#L64). Thanks.

jeevatkm commented on src/main/java/com/myjeeva/digitalocean/pojo/Volume.java:
```diff
@@ -0,0 +1,85 @@
+	private Integer sizeGigabytes;
+	
+	@SerializedName("created_at")
+	private Date createdAt;
```
Currently [this](https://github.com/jeevatkm/digitalocean-api-java/blob/master/src/main/java/com/myjeeva/digitalocean/pojo/Image.java#L58) name is followed for `created_at` attribute in all POJOs.

It's better to keep consistence.  

_PS: IMHO current variable name is nothing wrong_

jeevatkm commented on src/main/java/com/myjeeva/digitalocean/pojo/VolumeAction.java:
```diff
@@ -0,0 +1,93 @@
+	
+	@Expose
+	@SerializedName("size_gigabytes")
+	private Integer sizeGigabytes;
```
Please change datatype and variable name like [here](https://github.com/jeevatkm/digitalocean-api-java/blob/master/src/main/java/com/myjeeva/digitalocean/pojo/Image.java#L64). Thanks.

jeevatkm commented:
@strokine Thanks for complete implementation.  I have added a notes, please take care.

Also, I couldn't see any integration tests, could you please add it [here](https://github.com/jeevatkm/digitalocean-api-java/blob/master/src/test/java/com/myjeeva/digitalocean/DigitalOceanIntegrationTest.java)?

jeevatkm commented:
I forget to mention. Please apply Google Java Code Formatter, XML link is at [contributing section](https://github.com/jeevatkm/digitalocean-api-java#contributing). 

_PS: While applying code formatting in `enum` class, kindly select non-enum code lines and apply it otherwise it will spoil the enum alignments._

strokine commented:
Hey @jeevatkm , I've refactored the code as you suggested and added integration tests.
Before you run the IntTests you have to create 2 volumes, and provide ID from one, and Name from another. This way the IntTests should run better.

RE: Google Java Code Formatter, I don't have it set up on my machine, maybe you could just apply it yourself, or as an option you could use this plugin here? https://github.com/revelc/formatter-maven-plugin

Thanks for the library.
Eugene

jeevatkm commented:
@strokine Thanks again for PR.

I will take care -
* Code formatting
* `sizeGigabytes` -> `size`
* Javadoc
